qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -0,0 +1,250 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A Sampling Neural Network based on a given quantum circuit."""
+
+from typing import Tuple, Union, List, Callable, Any, Optional, Dict
+
+import numpy as np
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import Gradient, CircuitSampler, CircuitStateFn
+from qiskit.providers import BaseBackend, Backend
+from qiskit.utils import QuantumInstance
+
+from .sampling_neural_network import SamplingNeuralNetwork
+from ..exceptions import QiskitMachineLearningError
+
+
+class CircuitQNN(SamplingNeuralNetwork):
+    """A Sampling Neural Network based on a given quantum circuit."""
+
+    def __init__(self, circuit: QuantumCircuit,
+                 input_params: Optional[List[Parameter]] = None,
+                 weight_params: Optional[List[Parameter]] = None,
+                 interpret: Union[str, Callable[[Tuple[int, ...]], Any]] = 'tuple',
+                 dense: bool = False, output_shape: Union[int, Tuple[int, ...]] = None,
+                 gradient: Gradient = None,
+                 quantum_instance: Optional[Union[QuantumInstance, BaseBackend, Backend]] = None
+                 ) -> None:
+        """Initializes the Circuit Quantum Neural Network.
+
+        Args:
+            circuit: The (parametrized) quantum circuit that generates the samples of this network.
```

Comment:
Parameterized is in () as this optional, ie the circuit is a Circuit and can optionally be parameterized. If so then input_params and weight params needs to be defined, such that the total set of params there match what the circuit needs?

Reply:
Now parameterized is not in (). Circuit should be parameterized, otherwise it does not make much sense.

Reply:
actually it does make sense, you can use the circuit in a pytorch setting and train a classical NN based on the samples.